textile/features.textile

```diff
@@ -688,7 +688,9 @@ h3(#realtime-channel). RealtimeChannel
 ** @(RTL7b)@ Subscribe with a name argument and a listener argument subscribes a listener to only messages whose @name@ member matches the string name
 ** @(RTL7c)@ This clause has been replaced by "@RTL7g@":#RTL7g. It was valid up to and including specification version @3.0@.
 ** @(RTL7g)@ If the @attachOnSubscribe@ channel option is @true@, implicitly attaches the @RealtimeChannel@ if the channel is in the @INITIALIZED@, @DETACHING@, or @DETACHED@ states. The optional callback, if provided, is called according to "@RTL4d@":#RTL4d based on the implicit attach operation. The listener will always be registered regardless of the implicit attach result
-** @(RTL7h)@ If the @attachOnSubscribe@ channel option is @false@, the optional callback must not be called.
+** @(RTL7h)@ If the @attachOnSubscribe@ channel option is @false@, then the behaviour depends on the public API that a given SDK uses to communicate the result of an "@RTL7g@":#RTL7g implicit attach:
+*** If the SDK’s API accepts an optional callback (or equivalent, for example it provides two overloads of @#subscribe@, one which accepts such a callback and one which doesn’t) to communicate the result of an "@RTL7g@":#RTL7g implicit attach, then it is a programmer error to provide such a callback when the @attachOnSubscribe@ channel option is @false@. This programmer error should be handled in an idiomatic fashion (e.g. by halting the program).
```

Comment:
Do we want to throw exception for non optional callback here? Like when implicit attach is false

Reply:
Like, I am not sure, but what if adding listener becomes async operation when called through wrapper SDKs.

Reply:
We discussed this on a call and the above comments relate to believing that the optional callback communicates the result of the add-a-listener operation — after going through my explanation of why I believe this not to be the case (https://github.com/ably/specification/pull/208#discussion_r1790255796) we agreed that these comments can be resolved

Reply:
Yeah, now we have mutual understanding for the same and conversation can be closed.